diff --git a/third_party/blake3/blake3.BUILD.bazel b/third_party/blake3/blake3.BUILD.bazel
index 29b9299a..d67b384e 100644
--- a/third_party/blake3/blake3.BUILD.bazel
+++ b/third_party/blake3/blake3.BUILD.bazel
@@ -11,6 +11,7 @@ cc_library(
         "c/blake3.c",
         "c/blake3_dispatch.c",
         "c/blake3_portable.c",
+        "c/blake3_neon.c"
     ],
     hdrs = [
         "c/blake3.h",
