diff --git a/third_party/c_ares/config/darwin.h b/third_party/c_ares/config/darwin.h
--- a/third_party/c_ares/config/darwin.h
+++ b/third_party/c_ares/config/darwin.h
@@ -213,7 +213,7 @@
 /* Define to 1 if you have the AvailabilityMacros.h header file. */
 #define HAVE_AVAILABILITYMACROS_H 1
 
 /* Define to 1 if you have the MSG_NOSIGNAL flag. */
-#define HAVE_MSG_NOSIGNAL 1
+/* #undef HAVE_MSG_NOSIGNAL */
 
 /* Define to 1 if you have the <netdb.h> header file. */
 #define HAVE_NETDB_H 1
